OpenSourcePolitics/decidim-module-cleaner
# fix: CleanInactiveUsersJob fails on Active Record validation fail

#### Description 

If a user isn't valid, `CleanInactiveUsersJob` task will continuously fail, this is due to the command `Decidim::DestroyAccount` which attempts to invalidate_all_sessions before updating user's attributes.

#### Related to

* Notion : https://www.notion.so/opensourcepolitics/Toulouse-Decidim-cleaner-jobs-are-failing-9e74e006a00741b7b2b5fdf3c6433b3c?pvs=4

#### Testing

* Create multiple users with different `current_sign_in_at`
* Run command `bundle exec rake decidim_cleaner:clean_inactive_users`
* It should send email to concerned users

If you want to test the fix
* Force a user attribute to a value that does not pass validations : (nickname: "####")
* Run task
* It should run without errors
